src/phpDocumentor/Descriptor/File.php

```diff
@@ -0,0 +1,239 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.3
+ *
+ * @copyright 2010-2014 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Descriptor;
+
+use phpDocumentor\Reflection\DocBlock;
+
+/**
+ * Represents a file in the project.
+ */
+Final class File
+{
+    /**
+     * @var DocBlock | Null
+     */
+    private $docBlock = null;
+
+    /** @var string $hash */
+    private $hash;
+
+    /** @var string $name */
+    private $name = null;
+
+    /** @var string $path */
+    private $path = null;
+
+    /** @var string|null $source */
+    private $source = null;
+
+    /** @var Collection $namespaceAliases */
+    private $namespaceAliases;
+
+    /** @var File[] $includes */
+    private $includes = array();
+
+    /** @var Function_[] $functions */
+    private $functions = array();
+
+    /** @var Constant[] $constants */
+    private $constants = array();
+
+    /** @var Class_[] $classes */
+    private $classes = array();
+
+    /** @var Interface_[] $interfaces */
+    private $interfaces = array();
+
+    /** @var Trait_[] $traits */
+    private $traits = array();
+
+    /**
+     * Initializes a new file descriptor with the given hash of its contents.
+     *
+     * @param string $hash An MD5 hash of the contents if this file.
+     */
+    public function __construct($hash, $path, $source = null, DocBlock $docBlock = null)
+    {
+        $this->hash = $hash;
+        $this->path = $path;
+        $this->name = basename($path);
+        $this->source = $source;
+        $this->docBlock = $docBlock;
+    }
+
+    /**
+     * Returns the hash of the contents for this file.
+     *
+     * @return string
+     */
+    public function getHash()
+    {
+        return $this->hash;
+    }
+
+    /**
+     * Retrieves the contents of this file.
+     *
+     * @return string|null
+     */
+    public function getSource()
+    {
+        return $this->source;
+    }
+
+    /**
+     * Returns the namespace aliases that have been defined in this file.
+     *
+     * @return Collection
+     */
+    public function getNamespaceAliases()
+    {
+        return $this->namespaceAliases;
+    }
+
+    /**
+     * Returns a list of all includes that have been declared in this file.
+     *
+     * @return File[]
+     */
+    public function getIncludes()
+    {
+        return $this->includes;
+    }
+
+    /**
+     * @param File $include
+     */
+    public function addInclude(File $include)
```

Comment:
This should be a Path object because not all File objects are available in the FileAssembler and it makes caching and updating easier (if a user deletes a file we would otherwise have to iterate all File objects in the project and update them). The View Component in phpDocumentor or the template will have to ensure that a correct link is available when working with the templates